pkg/cmd/adm/restart.go

```diff
@@ -1,157 +1,200 @@
 		return err
 	}
 
-	ctx.Printlnf("The deployment was scaled back to '%d'", originalReplicas)
+	if len(olmDeploymentList.Items) == 0 {
+		ctx.Printlnf("No OLM based deployment restart happened as Olm deployment found in namespace %s is 0", ns)
+	} else {
+		for _, olmDeployment := range olmDeploymentList.Items {
+			ctx.Printlnf("Proceeding to delete the Pods of %v", olmDeployment)
+
+			if err := deleteAndWaitForPods(ctx, cl, olmDeployment, f, ioStreams); err != nil {
+				return err
+			}
+		}
+	}
+	if len(nonOlmDeploymentlist.Items) != 0 {
+		for _, nonOlmDeployment := range nonOlmDeploymentlist.Items {
+
+			ctx.Printlnf("Proceeding to restart the non-OLM deployment %v", nonOlmDeployment)
+
+			if err := restartNonOlmDeployments(ctx, nonOlmDeployment, f, ioStreams); err != nil {
+				return err
+			}
+			//check the rollout status
+			ctx.Printlnf("Checking the status of the rolled out deployment %v", nonOlmDeployment)
+			if err := checkRolloutStatus(ctx, f, ioStreams, "provider=codeready-toolchain"); err != nil {
+				return err
+			}
+		}
+	} else {
+		ctx.Printlnf("No Non-OLM based deployment restart happened as Non-Olm deployment found in namespace %s is 0", ns)
+	}
 	return nil
 }
 
-func restartHostOperator(ctx *clicontext.CommandContext, hostClient runtimeclient.Client, hostNamespace string) error {
-	deployments := &appsv1.DeploymentList{}
-	if err := hostClient.List(context.TODO(), deployments,
-		runtimeclient.InNamespace(hostNamespace),
-		runtimeclient.MatchingLabels{"olm.owner.namespace": "toolchain-host-operator"}); err != nil {
+func deleteAndWaitForPods(ctx *clicontext.CommandContext, cl runtimeclient.Client, deployment appsv1.Deployment, f cmdutil.Factory, ioStreams genericclioptions.IOStreams) error {
+	ctx.Printlnf("Listing the pods to be deleted")
+	//get pods by label selector from the deployment
+	pods := corev1.PodList{}
+	selector, _ := metav1.LabelSelectorAsSelector(deployment.Spec.Selector)
+	if err := cl.List(ctx, &pods,
+		runtimeclient.MatchingLabelsSelector{Selector: selector},
+		runtimeclient.InNamespace(deployment.Namespace)); err != nil {
 		return err
 	}
-	if len(deployments.Items) != 1 {
-		return fmt.Errorf("there should be a single deployment matching the label olm.owner.namespace=toolchain-host-operator in %s ns, but %d was found. "+
-			"It's not possible to restart the Host Operator deployment", hostNamespace, len(deployments.Items))
+	ctx.Printlnf("Starting to delete the pods")
+	//delete pods
+	for _, pod := range pods.Items {
+		pod := pod // TODO We won't need it after upgrading to go 1.22: https://go.dev/blog/loopvar-preview
+		if err := cl.Delete(ctx, &pod); err != nil {
+			return err
+		}
+
+		ctx.Printlnf("Checking the status of the deleted pod's deployment %v", deployment)
+		//check the rollout status
+		if err := checkRolloutStatus(ctx, f, ioStreams, "kubesaw-control-plane=kubesaw-controller-manager"); err != nil {
```

Comment:
Calling delete and then immediately rollout status can be tricky - the execution of the command can be too fast so the deletion of the pod doesn't have to be propagated and reflected anywhere, thus the rollout status can pass right away as the deployment looks still OKish.

Reply:
i have tried to do that out of delete method